File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.java
class Solution {
    public int minPairSum(int[] nums) {
        int result = 0, tmp = 0;
        
        Arrays.sort(nums);
        
        for (int i = 0, j = nums.length - 1; i < nums.length; i++) {
            tmp = nums[i] + nums[j];
            result = Math.max(result, tmp);
            j--;
        }
        
        return result;
    }
}

File: 1877-minimize-maximum-pair-sum-in-array/1877-minimize-maximum-pair-sum-in-array.kt
class Solution {
fun minPairSum(nums: IntArray): Int {
    nums.sort()
    var max = 0
    var i = 0
    var j = nums.size - 1
    while (i < j) {
        max = Math.max(max, nums[i++] + nums[j--])
    }
    return max
}
}